components/engine_traits/src/engines.rs

```diff
@@ -30,4 +52,47 @@ impl<K: KvEngine, R: RaftEngine> Engines<K, R> {
     pub fn sync_kv(&self) -> Result<()> {
         self.kv.sync()
     }
+
+    pub fn tablet(&self) -> &K {
+        if let Some(tablet) = &self.tablet_kv {
+            &tablet
+        } else {
+            &self.kv
+        }
+    }
+
+    pub fn load_tablet(&mut self, id: u64, suffix: u64) -> Result<()> {
+        let tablet = self.tablets.as_ref().unwrap().open_tablet(id, suffix)?;
+        self.tablet_kv = Some(tablet);
+        return Ok(());
+    }
+}
+
+impl<K: Clone, R: Clone> Clone for Engines<K, R> {
+    #[inline]
+    fn clone(&self) -> Engines<K, R> {
+        Engines {
+            kv: self.kv.clone(),
+            raft: self.raft.clone(),
+            tablets: if let Some(tablets) = &self.tablets {
```

Comment:
you can use `tablets.map(|x| x.clone())` to simplify the code here.

Reply:
Thanks.